confidential-data-hub/overlay-network/src/overlay_network/nebula.rs

```diff
@@ -0,0 +1,230 @@
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::{OverlayNetworkError, Result};
+use crate::overlay_network::config_templates::{WORKER_CONFIG_TEMPLATE, LIGHTHOUSE_CONFIG_TEMPLATE};
+use kms::{Annotations, ProviderSettings};
+use nix::ifaddrs::getifaddrs;
+use nix::sys::socket::{AddressFamily, SockaddrLike};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use serde_yml;
+use std::fs;
+use std::net::Ipv4Addr;
+use std::process::Command;
+
+pub struct NebulaMesh {
+    pod_name: String,
+    lighthouse_ip: String,
+}
+
+// FIXME This should be a shared struct, if possible, with trustee's nebula
+//       plugin. It's that plugin's custom protocol.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NebulaPluginResponse {
+    pub node_crt: Vec<u8>,
+    pub node_key: Vec<u8>,
+    pub ca_crt: Vec<u8>,
+}
+
+const NEBULA_BIN: &str = "/opt/overlay-network/nebula";
+
+const CA_CERT_PATH: &str = "/tmp/nebula/ca.crt";
+const POD_CERT_PATH: &str = "/tmp/nebula/pod.crt";
+const POD_KEY_PATH: &str = "/tmp/nebula/pod.key";
+const LIGHTHOUSE_CONFIG_PATH: &str = "/tmp/nebula/lighthouse-config.yaml";
+const WORKER_CONFIG_PATH: &str = "/tmp/nebula/config.yaml";
+
+// FIXME These should be configurable
+const LIGHTHOUSE_IP: Ipv4Addr = Ipv4Addr::new(192, 168, 100, 100);
+const LIGHTHOUSE_MASK: Ipv4Addr = Ipv4Addr::new(255, 255, 255, 0);
+
+impl NebulaMesh {
+    /// Initialize a nebula mesh. The general approach is as follows:
+    /// - Calculate what the mesh IP will be for this worker.
+    /// - Ask trustee for its nebula credentials
+    /// - Start the nebula daemon.
+    pub async fn init(&self) -> Result<()> {
+        let is_lighthouse: bool = self.lighthouse_ip.is_empty();
+
+        let (mesh_ip, which_config);
```

Comment:
to be more rust-like, you could do `let (mesh_ip, which_config) if is_lighthouse {` and then have each arm return a tuple